pkg/ottl/LANGUAGE.md

```diff
@@ -2,6 +2,15 @@
 
 The OTTL grammar includes function invocations, Values and Boolean Expressions. These parts all fit into a Statement, which is the basis of execution in the OTTL.
 
+### Design principles
+
+OTTL is intended as a domain-specific language (DSL) for telemetry mutation and generation,
+and is not intended to be used as a general-purpose programming language.
+
+OTTL has been designed to work directly with [pdata](https://github.com/open-telemetry/opentelemetry-collector/tree/main/pdata).
+While OTTL can work with external data formats due to being implemented directly in Go instead of on top of a custom VM,
```

Comment:
@TylerHelmuth What do you think about this statement in light of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161? Is it okay for us to have contexts that work pretty much exclusively with external data formats even if it's not "expressly intended", or would we contradict ourselves with those contexts?

Reply:
I think this is too aggressive of a statement.  I want to leave some wiggle room to handle collector situations like #25161.  Currently OTTL's framework is still very pdata agnostic and the places where we do make direct references to pdata, like some getters/setters, aren't limiting how other people use OTTL.